### PR TITLE
fix(userspace): safely check string bounded access

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

/area engine

**What this PR does / why we need it**:

This prevents out-of-bound string access from happening.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by FedeDP — APPROVED** [review]
/approve

File: userspace/engine/rule_loader.cpp [hunk 1]
```diff
@@ -223,44 +225,48 @@ std::string rule_loader::context::snippet(const falco::load_result::rules_conten
 	// However, some lines can be very very long, so the walk
 	// forwards/walk backwards is capped at a maximum of
 	// snippet_width/2 characters in either direction.
-	for(; from > 0 && snip_content.at(from) != '\n' && (loc.pos.pos - from) < (snippet_width/2); from--);
+	size_t from = pos;
+	for(; from > 0 && snip_content.at(from) != '\n' && (pos - from) < (snippet_width/2); from--);
 
-	size_t to = loc.pos.pos;
-	for(; to < snip_content.size()-1 && snip_content.at(to) != '\n' && (to - loc.pos.pos) < (snippet_width/2); to++);
+	size_t to = pos;
```
Comment (Andreagit97) [inline userspace/engine/rule_loader.cpp:231]: `to` seems to have a different value than before these changes, now we set the value after the for loop, is that correct?

**Review by Andreagit97 — COMMENTED** [review]
just a minor comment

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**